src/nvim/window.c

```diff
@@ -1010,6 +1010,10 @@ void ui_ext_win_position(win_T *wp, bool validate)
       comp_col += grid->comp_col;
       comp_row = MAX(MIN(comp_row, Rows - wp->w_height_outer - (p_ch > 0 ? 1 : 0)), 0);
       comp_col = MAX(MIN(comp_col, Columns - wp->w_width_outer), 0);
+      int right_extra = Columns - (int)c.col - wp->w_width - (int)strlen(c.border_chars[2]);
```

Comment:
no strlen, just check whether `c.border_chars[2][0] == NUL)` (width of border is always one)

Reply:
done.